// karma-fixture/lib/fixture.js
// Generated by CoffeeScript 1.9.1
(function() {
  var Fixture,
    slice = [].slice;

  Fixture = (function() {
    function Fixture(base, id) {
      this.base = base != null ? base : 'spec/fixtures';
      this.id = id != null ? id : 'fixture_container';
      this.json = [];
      this.el = window[this.id] || ((function(_this) {
        return function() {
          var container;
          container = document.createElement('div');
          container.setAttribute('id', _this.id);
          return document.body.appendChild(container);
        };
      })(this))();
    }

    Fixture.prototype.load = function() {
      var append, err, filename, filenames, j, json, k, len, results, string;
      filenames = 2 <= arguments.length ? slice.call(arguments, 0, j = arguments.length - 1) : (j = 0, []), append = arguments[j++];
      if (append == null) {
        append = false;
      }
      if (typeof append !== 'boolean') {
        filenames.push(append);
        append = false;
      }
      if (append === false) {
        this.cleanup();
      }
      results = [];
      for (k = 0, len = filenames.length; k < len; k++) {
        filename = filenames[k];
        string = typeof __html__ !== "undefined" && __html__ !== null ? __html__[this.base + "/" + filename] : void 0;
        if (string == null) {
          this._throwNoFixture();
        }
        if (filename.indexOf('.json') !== -1) {
          try {
            json = JSON.parse(string);
            this.json.push(json);
            results.push(json);
          } catch (_error) {
            err = _error;
          }
        } else {
          results.push(this._appendFixture(string));
        }
      }
      if (results.length === 1) {
        results = results[0];
      }
      return results;
    };

    Fixture.prototype.set = function() {
      var append, j, k, len, results, string, strings;
      strings = 2 <= arguments.length ? slice.call(arguments, 0, j = arguments.length - 1) : (j = 0, []), append = arguments[j++];
      if (append == null) {
        append = false;
      }
      if (typeof append !== 'boolean') {
        strings.push(append);
        append = false;
      }
      if (append === false) {
        this.cleanup();
      }
      results = [];
      for (k = 0, len = strings.length; k < len; k++) {
        string = strings[k];
        results.push(this._appendFixture(string));
      }
      if (results.length === 1) {
        results = results[0];
      }
      return results;
    };

    Fixture.prototype.cleanup = function() {
      this.json = [];
      return this.el.innerHTML = '';
    };

    Fixture.prototype._appendFixture = function(html_string) {
      var i, results, temp_div;
      temp_div = document.createElement('div');
      temp_div.innerHTML = html_string;
      results = [];
      while (i = temp_div.firstChild) {
        if (i.nodeType !== 1) {
          temp_div.removeChild(i);
        } else {
          this.el.appendChild(i);
          results.push(i);
          if (i.nodeName === 'SCRIPT') {
            eval(i.innerText);
          }
        }
      }
      return results;
    };

    Fixture.prototype._throwNoFixture = function() {
      throw new ReferenceError('Fixture not found');
    };

    return Fixture;

  })();

  if (typeof exports === 'object') {
    module.exports = Fixture;
  } else if (typeof define === 'function' && define.amd) {
    define('fixture', [], function() {
      return Fixture;
    });
  } else {
    this['Fixture'] = Fixture;
  }

}).call(this);
